mypy/checkexpr.py

```diff
@@ -2125,7 +2125,11 @@ def visit_comparison_expr(self, e: ComparisonExpr) -> Type:
             method_type = None  # type: Optional[mypy.types.Type]
 
             if operator == 'in' or operator == 'not in':
-                right_type = self.accept(right)  # always validate the right operand
+                # If the right operand has partial type, look it up without triggering
+                # a "Need type annotation ..." message, as it would be noise.
+                right_type = self.find_partial_type_ref_fast_path(right)
+                if right_type is None:
+                    right_type = self.accept(right)  # Validate the right operand
```

Comment:
What if we still defer (when possible) the current function, will it fix the `--strict-equality` problem?

Reply:
I tried this but it didn't help. I think that the reason was that we don't infer more precise types after deferral.